libstuff/SRandom.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <random>
+using namespace std;
+
+// Random number generator class.
+class SRandom {
+private:
+    static mt19937_64* generator;
+    static uniform_int_distribution<uint64_t>* distribution64;
```

Comment:
Why use pointers here rather than just static instances?  The only argument I can think of is if you want to preserve the ability for this random initializer to be used *before* `main()` is called (eg, in a global scope constructor), and thus potentially before the global attributes have themselves been initialized.

However, in general `libstuff` isn't safe to to use in a global scope constructor because `SInitialize()`, and that isn't called until `main()`.  But that's fine because logic that runs in a global scope constructor is very dangerous (due to it executing in an unpredictable order), so we in general shouldn't be doing it anyway.

The upshot is, a simpler solution would just use static variables, and barring some compelling reason to go with a more complex solution, I vote we keep it simple.